File: Estructura_de_datos_JavaScript/mergeSort.js
function mergeSort(arr) {
    if (arr.length <= 1) return arr;

    const medio = Math.floor(arr.length / 2);
    const izquierda = arr.slice(0, medio);
    const derecha = arr.slice(medio);

    return merge(mergeSort(izquierda), mergeSort(derecha));
}

function merge(izquierda, derecha) {
    let resultado = [];
    let i = 0, j = 0;

    while (i < izquierda.length && j < derecha.length) {
        if (izquierda[i] <= derecha[j]) {
            resultado.push(izquierda[i++]);
        } else {
            resultado.push(derecha[j++]);
        }
    }

    return resultado.concat(izquierda.slice(i)).concat(derecha.slice(j));
}

let numeros = [34, 7, 23, 32, 5, 62];
console.log("Original:", numeros);
numeros = mergeSort(numeros);
console.log("Ordenado:", numeros);

File: Estructura_de_datos_Python/mergeSort.py
def mergeSort(arr):
    if len(arr) > 1:
        medio = len(arr) // 2
        izquierda = arr[:medio]
        derecha = arr[medio:]

        mergeSort(izquierda)
        mergeSort(derecha)

        i = j = k = 0

        while i < len(izquierda) and j < len(derecha):
            if izquierda[i] <= derecha[j]:
                arr[k] = izquierda[i]
                i += 1
            else:
                arr[k] = derecha[j]
                j += 1
            k += 1

        while i < len(izquierda):
            arr[k] = izquierda[i]
            i += 1
            k += 1

        while j < len(derecha):
            arr[k] = derecha[j]
            j += 1
            k += 1

numeros = [34, 7, 23, 32, 5, 62]
print("Original:", numeros)
mergeSort(numeros)
print("Ordenado:", numeros)
